src/apps/service/apps/jobs/submissions/async.clj

```diff
@@ -0,0 +1,118 @@
+            [apps.util.config :as config]))
+
+(defn- max-batch-paths-exceeded
+  [max-paths first-list-path first-list-count]
+  (throw+
+    {:type       :clojure-commons.exception/illegal-argument
+     :error      (str "The HT Analysis Path List exceeds the maximum of "
+                      max-paths
+                      " allowed paths.")
+     :path       first-list-path
+     :path-count first-list-count}))
+
+(defn- validate-path-lists
+  [path-lists]
+  (let [[first-list-path first-list] (first path-lists)
+        first-list-count             (count first-list)]
+    (when (> first-list-count (config/path-list-max-paths))
+      (max-batch-paths-exceeded (config/path-list-max-paths) first-list-path first-list-count))
+    (when-not (every? (comp (partial = first-list-count) count second) path-lists)
+      (throw+ {:type  :clojure-commons.exception/illegal-argument
+               :error "All HT Analysis Path Lists must have the same number of paths."}))
+    path-lists))
+
+(defn- get-path-list-contents
+  [user path]
+  (try+
+    (when (seq path) (data-info/get-path-list-contents user path))
+    (catch Object _
+      (log/error (:throwable &throw-context)
+                 "job submission failed: Could not get file contents of path list input.")
+      (throw+))))
+
+(defn- get-path-list-contents-map
+  [user paths]
+  (into {} (map (juxt identity (partial get-path-list-contents user)) paths)))
+
+(defn- map-slice
+  [m n]
+  (->> (map (fn [[k v]] (vector k (nth v n))) m)
+       (into {})
+       (remove-nil-values)))
+
+(defn- map-slices
+  [m]
+  (let [max-count (apply max (map (comp count val) m))]
+    (mapv (partial map-slice m) (range max-count))))
+
+
+(defn- substitute-param-values
+  [path-map config]
+  (->> (map (fn [[k v]] (vector k (get path-map v v))) config)
+       (into {})))
+
+(defn- format-submission-in-batch
+  [submission job-number path-map]
+  (let [job-suffix (str "analysis-" (inc job-number))]
+    (assoc (update-in submission [:config] (partial substitute-param-values path-map))
+      :group      (config/jex-batch-group-name)
+      :name       (str (:name submission) "-" job-suffix)
+      :output_dir (ft/path-join (:output_dir submission) job-suffix))))
+
+(defn- submit-job-in-batch
+  [apps-client user submission job-number path-map]
+  (try+
+    (submit/submit-and-register-private-job apps-client
+                                            user
+                                            (format-submission-in-batch submission job-number path-map))
+    (catch Object _
+      (log/error (:throwable &throw-context)
+                 "batch job submission failed.")
+      {:status jp/failed-status})))
+
+(defn- preprocess-batch-submission
+  [submission output-dir parent-id]
+  (assoc submission
+    :output_dir           output-dir
+    :parent_id            parent-id
+    :create_output_subdir false))
+
+(defn submit-batch-jobs
```

Comment:
All of the functions in this namespace were imported from `apps.service.apps.jobs.submissions`, but this function is slightly different than the old batch submission function.

This function wraps sub-job submissions in a `future` call, but it also catches all exceptions so it can mark the parent job as `Failed`, and it will also send notifications to the user on success or failure.